waves/src/App.tsx

```diff
@@ -243,19 +249,27 @@ function App() {
                             />
                             <Center w="10px">
                                 <Box zIndex={2}>
-                                    <ExchangeIcon dispatch={dispatch} />
+                                    <ExchangeIcon
+                                        onClick={() =>
+                                            dispatch({
+                                                type: "SwapAssetTypes",
+                                                value: {
+                                                    betaAmount,
+                                                },
+                                            })}
+                                    />
```

Comment:
Personally I prefer extracting functions for this which makes the html code more readable.

Reply:
I am not sure I fully agree. I find React code easier to read if components are extracted in a particular way. It is quite nice to know that clicking this button will swap the assets _without_ having to go into the component and check what it does.

This way, all the state and how / when it is modified is within a single file. If you start extracting functions for this, you have to jump around again between fn definition vs UI definition.

Reply:
One way to make this more concise is to extract helper functions for creating actions as in:

```ts
function swapAssetTypes(betaAmount: number) {
	// ...
}
```

so you can do:

```ts
dispatch(swapAssetTypes(betaAmount))
```

However, that is also quite the boilerplate esp. if things are still changing a lot so :man_shrugging:

Reply:
I'd probably first go ahead and remove the `.value` from all actions and just embed whatever type you want directly. That would also remove 2 lines of code from creating this action.